fly/commands/get_team.go

```diff
@@ -0,0 +1,58 @@
+package commands
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/concourse/concourse/atc"
+	"github.com/concourse/concourse/fly/rc"
+	yaml "gopkg.in/yaml.v2"
+)
+
+type GetTeamCommand struct {
+	Team string `short:"n" long:"team" required:"true" description:"Get configuration of this team"`
+	JSON bool   `short:"j" long:"json" description:"Print config as json instead of yaml"`
+}
+
+func (command *GetTeamCommand) Execute(args []string) error {
+	asJSON := command.JSON
+	teamName := command.Team
+
+	target, err := rc.LoadTarget(Fly.Target, Fly.Verbose)
+	if err != nil {
+		return err
+	}
+
+	if err := target.Validate(); err != nil {
+		return err
+	}
+
+	config, found, err := target.Team().Config(teamName)
+	if err != nil {
+		return err
+	}
+
+	if !found {
+		return errors.New("team not found")
+	}
+
+	return dumpTeam(config, asJSON)
+}
+
+func dumpTeam(config atc.Team, asJSON bool) error {
+	var payload []byte
+	var err error
+	if asJSON {
+		payload, err = json.Marshal(config)
+	} else {
+		payload, err = yaml.Marshal(config)
```

Comment:
I think it's a little weird that this shows the internal team ID since we don't currently expose that anywhere else. I don't feel too strongly about it though. This is the first `get-X` command that acts on an API resource as opposed to pipeline config so it just hasn't come up, I suppose.

Reply:
Thanks for your feedback!

The output format of `get-team` command followed `teams`, not `get-pipeline`.

Reply:
Now, this command, `get-team`, become a filter to get a specified team from the output of `teams` as planned.

Reply:
I think it was OK before, it just felt weird to include the ID. 🙂 Happy to merge either way, this is something we can easily iterate on as the use case evolves.

Reply:
I referenced `fly teams` command.
This command's output is below. (with `--json`)

<details>
<summary>output</summary>
<pre>
<code>
fly -t lm teams --json      
[
  {
    "id": 3,
    "name": "mechanic",
    "auth": {
      "owner": {
        "groups": [],
        "users": [
          "local:mechanic"
        ]
      }
    }
  }
]
</code>
</pre>
</details>

If we should hide the ID, `get-team` and `teams` commands are fixed together.

> Happy to merge either way, this is something we can easily iterate on as the use case evolves.

If we can easily iterate on as the use case evolves, I want to complete changes in this PR. 😁

Is it OK?

Reply:
Showing the ID in `--json` is fine since that shows all the internal details - I only thought it was confusing in the default case (YAML). Not a big deal either way - I'll go ahead and merge. :)